[lldb/Process] Silence toolchain mismatch warnings for Scripted Processes (NFC)

medismailben opened the pull request:
When using Scripted Processes, it's very likely that lldb's version
might not align with the swift compiler that built the target.

Thus, this will cause the toolchain mismatch warnings to be shown to the
user every time they "stop" in a Swift frame.

Since we shouldn't need Swift's AST to symbolicate the stopped stackframe,
we can safely silence these warnings for Scripted Processes.

This PR also introduces the ability to load all images used in a crashlog, which
allow the underlying Scripted Process to create artificial frames for non-crashed threads.

rdar://90396265

Signed-off-by: Med Ismail Bennani <medismail.bennani@gmail.com>

medismailben commented:
@swift-ci test

JDevlieghere commented on lldb/source/Target/Process.cpp:
```diff
@@ -5930,6 +5930,11 @@ void Process::PrintWarningCantLoadSwiftModule(const Module &module,
 
 void Process::PrintWarningToolchainMismatch(const SymbolContext &sc) {
+  if (GetTarget().GetProcessLaunchInfo().IsScriptedProcess())
+    // It's very likely that the debugger used to launch the ScriptedProcess
+    // will not match the compiler version used to build the target, so we
+    // should just silence the toolchain mismatch warning.
```
This comment doesn't really explain why this is necessary. I would explain that there's no need for the debugger and swift compiler to match, because symbolicating a crashlog doesn't require the serialized AST. I would also explain why it's safe to silence this warning: because it doesn't depend on us using that, but instead this function is called from a relatively generic place (`FrameSelectedCallback`).

medismailben committed:
[lldb/crashlog] Create artificial frames for non-crashed scripted threads

This patch pipes down the `-a|--load-all` crashlog command option to the
Scripted Process initializer to load all the images used by crashed
process instead of only loading the images related to the crashed
thread.

This allows us to recreate artificial frames also for the non-crashed
scripted threads.

rdar://90396265

Differential Revision: https://reviews.llvm.org/D121826

Signed-off-by: Med Ismail Bennani <medismail.bennani@gmail.com>

medismailben committed:
[lldb/Process] Silence toolchain mismatch warnings for Scripted Processes (NFC)

When using Scripted Processes, it's very likely that lldb's version
might not align with the swift compiler that built the target.

Thus, this will cause the toolchain mismatch warnings to be shown to the
user every time they "stop" in a Swift frame.

Since we shouldn't need Swift's AST to symbolicate the stopped stackframe,
we can safely silence these warnings for Scripted Processes.

Signed-off-by: Med Ismail Bennani <medismail.bennani@gmail.com>

medismailben commented:
@swift-ci test
